src/main/java/org/prebid/server/analytics/AnalyticsReporterDelegator.java

```diff
@@ -72,4 +90,79 @@ public AnalyticsReporterDelegator(List<AnalyticsReporter> delegates,
+                ? analytics.fieldNames() : null;
+
+        if (analyticsFieldNames != null) {
+            final List<String> unknownAdapterNames = StreamSupport.stream(
+                    Spliterators.spliteratorUnknownSize(analyticsFieldNames, Spliterator.ORDERED), false)
+                    .filter(adapter -> !reporterNames.contains(adapter))
+                    .collect(Collectors.toList());
+            if (CollectionUtils.isNotEmpty(unknownAdapterNames)) {
+                logger.warn(
+                        String.format("Unknown adapters in ext.prebid.analytics[].adapter: %s", analyticsFieldNames));
+            }
+        }
+    }
+
+    private static <T> T updateEvent(T event, String delegatorAdapter) {
+        if (!ADAPTERS_PERMITTED_FOR_FULL_DATA.contains(delegatorAdapter) && event instanceof AuctionEvent) {
+            final AuctionEvent auctionEvent = (AuctionEvent) event;
+            final AuctionContext updatedAuctionContext =
+                    updateAuctionContextForDelegator(auctionEvent.getAuctionContext(), delegatorAdapter);
+            return updatedAuctionContext != null
+                    ? (T) auctionEvent.toBuilder().auctionContext(updatedAuctionContext).build()
+                    : event;
+        }
+
+        return event;
+    }
+
+    private static AuctionContext updateAuctionContextForDelegator(AuctionContext context, String delegatorAdapter) {
+        final BidRequest updatedBidRequest = updateBidRequest(context.getBidRequest(), delegatorAdapter);
+
+        return updatedBidRequest != null ? context.toBuilder().bidRequest(updatedBidRequest).build() : null;
+    }
+
+    private static BidRequest updateBidRequest(BidRequest bidRequest, String adapterName) {
+        final ExtRequest requestExt = bidRequest.getExt();
+        final ExtRequestPrebid extPrebid = requestExt != null ? requestExt.getPrebid() : null;
+        final ObjectNode analytics = extPrebid != null ? extPrebid.getAnalytics() : null;
+        ObjectNode preparedAnalytics = null;
+        if (analytics != null && !analytics.isEmpty()) {
+            preparedAnalytics = prepareAnalytics(analytics, adapterName);
+        }
+        final ExtRequest updatedExtRequest = preparedAnalytics != null ? ExtRequest.of(extPrebid.toBuilder()
+                .analytics(preparedAnalytics)
+                .build()) : null;
+
+        if (updatedExtRequest != null) {
+            updatedExtRequest.addProperties(requestExt.getProperties());
+            return bidRequest.toBuilder().ext(updatedExtRequest).build();
+        }
+
+        return null;
+    }
+
+    private static ObjectNode prepareAnalytics(ObjectNode analytics, String adapterName) {
+        final ObjectNode analyticsNodeCopy = analytics.deepCopy();
+        analyticsNodeCopy.fieldNames().forEachRemaining(fieldName -> {
+            if (!Objects.equals(adapterName, fieldName)) {
+                analyticsNodeCopy.remove(fieldName);
```

Comment:
We don't need to iterate through each node but just override existing `analytics` node internals with `adapterName` node.